src/core_ocean/shared/mpas_ocn_vmix_gotm.F

```diff
@@ -0,0 +1,531 @@
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  ocn_vmix_gotm
+!
+!> \brief MPAS ocean vertical mixing interface to GOTM
+!> \author Qing Li
+!> \date   11 May 2020
+!> \details
+!>  This module contains the routines for calls into GOTM
+!>
+!
+!-----------------------------------------------------------------------
+
+module ocn_vmix_gotm
+
+   use mpas_derived_types
+   use mpas_pool_routines
+   use mpas_timer
+   use mpas_constants
+   use mpas_log
+   use mpas_timekeeping
+
+   use ocn_constants
+   use ocn_config
+   use ocn_diagnostics_variables
+
+   ! TODO: read in GOTM namelist for testing <20200512, Qing Li> !
+   use mpas_io_units
+
+   use turbulence, only: init_turbulence, do_turbulence, clean_turbulence
+   use turbulence, only: gotm_vertVisc=>num, gotm_vertDiffT=>nuh, &
+                         gotm_vertDiffS=>nus, gotm_TKE=>tke,      &
+                         gotm_eps=>eps, gotm_L=>L,                &
+                         gotm_kb=>kb, gotm_epsb=>epsb
+   use mtridiagonal, only: init_tridiagonal
+
+   implicit none
+   private
+   save
+
+   !--------------------------------------------------------------------
+   !
+   ! Public parameters
+   !
+   !--------------------------------------------------------------------
+
+   !--------------------------------------------------------------------
+   !
+   ! Public member functions
+   !
+   !--------------------------------------------------------------------
+
+   public :: ocn_vmix_coefs_gotm_build, &
+             ocn_vmix_gotm_init,        &
+             ocn_vmix_gotm_finalize
+
+   !--------------------------------------------------------------------
+   !
+   ! Private module variables
+   !
+   !--------------------------------------------------------------------
+
+   logical :: gotmOn
+   real (kind=RKIND) :: gotm_dt
+   real (kind=RKIND) :: gotmBottomDragCoef, gotmBottomRoughnessLength, &
+                        gotmSurfaceRoughnessLength
+
+
+!***********************************************************************
+
+contains
+
+!***********************************************************************
+!
+!  routine ocn_vmix_coefs_gotm_build
+!
+!> \brief   Computes mixing coefficients using GOTM
+!> \author  Qing Li
+!> \date    11 May 2020
+!> \details
+!>  This routine computes the vertical mixing coefficients for momentum
+!>  and tracers by calling GOTM routines.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_vmix_coefs_gotm_build(meshPool, statePool, forcingPool, err, timeLevelIn)!{{{
```

Comment:
@mattdturner, I rebased this from last September, when we passed the meshPool in. We should add
```
use ocn_mesh
```
at the top and remove all meshPool variable declarations and lines with
```
mpas_pool_get_dimension(meshPool
```
correct? @qingli411 let's wait on that step until after you have successfully verified your test case with the current code.

Reply:
Correct, you would need to add `use ocn_mesh`, remove all meshPool variable declarations (or you'll get unassociated pointer errors), and all calls to `mpas_pool_get_dimension(meshPool)`.

Reply:
And also remember the changes to account for the changes in nCells, nCellsArray, nCellsSolve (to nCellsAll, nCellsHalo(n-1), nCellsOwned) and equivalent changes to the nEdges, nVertex analogues.